packages/rooks/src/hooks/useHighlight.ts

```diff
@@ -0,0 +1,35 @@
+/**
+ * useHighlight
+ * @description search for a keyword in the DOM and highlight them
+ * @see {@link https://rooks.vercel.app/docs/useHighlight}
+ */
+import { useEffect, useRef } from "react";
+
+function useHighlight({
+  isReady = false,
+  keyword = "",
+}: {
+  isReady: boolean;
+  keyword: string;
+}) {
+  const ref = useRef<HTMLElement>(null);
+
+  useEffect(() => {
+    if (keyword && isReady && ref.current) {
+      const contentElement = ref.current;
+      if (contentElement) {
+        const text = contentElement.innerHTML;
+        const regex = new RegExp(keyword, "g");
+        const newText = text.replace(
+          regex,
+          `<mark class="highlighted">${keyword}</mark>`
```

Comment:
One thing I would like to add in the tests is that it should not highlight if it is already highlighted. For eg:
✅
`something keyword` => `something <mark class="highlighted">keyword</mark>` 
❌
`something <mark class="highlighted">keyword</mark>` => `something <mark class="highlighted"><mark class="highlighted">keyword</mark></mark>`